### ISSUE TITLE
Test library with MSL v.4.0.0

### ISSUE THREAD

**Issue by mwetter** [lbl-srg/modelica-buildings#1789]
This issue is to test the `Buildings` Library with `MSL 4.0.0`.
The development branch is `issue1789_msl_4.0.0_beta-v3` and the tests start with https://github.com/mwetter/ModelicaStandardLibrary

**Comment by mwetter** [comment]
The conversion script changes in `Buildings/Examples/VAVReheat/Controls/ModeSelector.mo` the code
```
  Modelica.StateGraph.InitialStep initialStep
    annotation (Placement(transformation(extent={{-80,20},{-60,40}})));
```
to
```
  Modelica.StateGraph.InitialStep initialStep(nIn=1, nOut=1)
    annotation (Placement(transformation(extent={{-80,20},{-60,40}})));
```
The `nin=1` is wrong and causes an error in Dymola 2020x. This is corrected in 20d0057e9ba7d1471c1fc387a8d7fb6d7864856b

Also, it turns out that MSL 4.0.0 now does not tolerate the *errorneous* declaration
```
  State morPreCoo(                                                        nIn=2,
    mode=Buildings.Examples.VAVReheat.Controls.OperationModes.unoccupiedPreCool,
    nOut=1) "Pre-cooling mode"
    annotation (Placement(transformation(extent={{-40,-140},{-20,-120}})));
```
in the same model. This is now corrected to `nIn=1`.

**Comment by mwetter** [comment]
The current development branch is `issue1789_msl_4.0.0_beta-v3`. Here is a list of changes to update the library:
```bash
git checkout -b issue1789_msl_4.0.0_beta-v3
dymola package.mo &     # and convert the library
git commit -m "Applied conversion script from master" .
# Set Set initialStep(nIn=0... in ModeSelector.mo
git commit -m "Set initialStep(nIn=0... in ModeSelector.mo" -m "For #1789" .
rm -rf Media && git checkout issue1789_msl_4.0.0_beta Media
git commit -m "Run rm -rf Media && git checkout issue1789_msl_4.0.0_beta Media" -m "For 1789" Media/
git checkout issue1789_msl_4.0.0_beta-v2 Fluid/HeatExchangers/BaseClasses/ntu_epsilonZ.mo 
git commit -m "Run git checkout issue1789_msl_4.0.0_beta-v2 Fluid/HeatExchangers/BaseClasses/ntu_epsilonZ.mo" -m "For #1789" .
git checkout issue1789_msl_4.0.0_beta-v2 Utilities/Psychrometrics/Functions/TDewPoi_pW.mo 
git commit -m "Run issue1789_msl_4.0.0_beta-v2 Utilities/Psychrometrics/Functions/TDewPoi_pW.mo" -m "For #1789" Utilities/
find . -name Internal.mo -exec rm {} \;
# Update the package.order files
git commit -m "Run find . -name Internal.mo -exec rm {} \; and rewrote package.order" -m "For 1789" .
```
